dcos/client.go

```diff
@@ -101,6 +101,16 @@ func NewClientWithConfig(config *Config) (*APIClient, error) {
 	return c, nil
 }
 
+// CurrentConfig returns the *Configuration used by the client
+func (a *APIClient) CurrentConfig() *Configuration {
```

Comment:
This was not clear to me if it returns the `Configuration` or our DC/OS `Config`, does it work for your use-case to return the DC/OS config?

Right now we are in this situation with 2 config structs, as it was quickly when bootstraping the client-go code, though in the long-run we should just stick to a single config struct. So this method will likely break or its API will change in the future.

Reply:
Yeah you're right... i'll try that

Reply:
Oh wait.. we're not even storing the config. This is bad... 

think we need a chat about the config stuff.

I totally agree with you this should be merged into a single struct ( or maybe better interface? ) 

I mean `Configuration` totally does what I need for marathon. However I also agree with you that this is not the best situation.

The question is how to handle config changes in the users environment. @bamarni could we have a quick chat this week about Config and Configmanager? Maybe my thinking is too far but I thought of having an Config interface which could either implement a static Config or receive a more "dynamic" one via ConfigManager?